### ISSUE TITLE
Parser allows discriminator for inline object declaration

### ISSUE THREAD

**Issue by deiteris** [aml-org/amf#1452]
Your issue may already be reported! Please search on Github issues before creating one.

* **I'm submitting a ...**
  - [x] bug report
  - [ ] feature request
  - [ ] question
  - [ ] other

* **What is the current behavior?**

The parser successfully validates `discriminator` placed inside the inline declaration of the object type, while this is against the current specification.
https://github.com/raml-org/raml-spec/blob/master/versions/raml-10/raml-10.md#using-discriminator

* **If the current behavior is a bug, please provide the minimal steps to reproduce with AMF code. Please don't upload AMF code mixed with yours.**

The following RAML file reproduces the issue:
```yaml
#%RAML 1.0
title: Pets API

types:
  Pet:
    type: object
    properties:
      stats:
        type: object
        discriminator: health # Should not be valid
        properties:
          health: string

/pet:
  post:
    body:
      application/json:
        type: object
        discriminator: kind # Should not be valid
        properties:
          kind: string
```

* **What is the expected behavior?**

`discriminator` properties should not be allowed in inline declarations.

* **Please tell us about your environment:**
  
  - AMF Version: 5.0.9
  - AMF Distribution: JVM and JS
  - Operating System: Windows 10 21H2

**Comment by hghianni** [comment]
Hi @deiteris, this is a bug. I have created an internal [ticket](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001080mFYAQ/view) to fix it.
We will keep you posted via this github thread. Thanks!

**Comment by arielmirra** [comment]
Hi @deiteris! This has been fixed in PR #1747 and will be available in the next release of AMF. Please note we don't throw a violation but a warning instead in the case that we didn't checked before, this is to avoid breaking compatibility. Thank you!
